common/src/com/thoughtworks/go/config/materials/git/GitMaterial.java

```diff
@@ -235,7 +235,13 @@ public void afterCompletion(int status) {
                 });
             }
             int cloneDepth = shallowClone ? preferredCloneDepth : Integer.MAX_VALUE;
-            int returnValue = gitCommand.cloneFrom(outputStreamConsumer, url.forCommandline(), cloneDepth);
+            int returnValue;
+            if(executionContext.isServer()) {
+                returnValue = gitCommand.serverSideClone(outputStreamConsumer, url.forCommandline());
```

Comment:
> I am wondering if there is any way that checkout on server can be still executed?

Yes, If the material is a configuration repository. So this feature will only work for materials that don't store configuration.